### PR TITLE
noissue: converted Person.java to record. Moved adapter logic between…

### PR DESCRIPTION
**Pull request by torarnet**
… Person and CristinPerson to own class

Based on feedback from @StigNorland i decided to try and refactor Person model to record. This is the result.

I don't know if using a record is better, but at least I have it in a working state where it is now a record.

I also moved adapter logic to own class for conversion between Person and CristinPerson

Doing this as a draft PR for the time being, and want your feedback

### CODE REVIEW

**Review by brinxmat — COMMENTED** [review]
Seems tidier :)

**Comment by torarnet** [comment]
Ready for review

**Review by brinxmat — APPROVED** [review]

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java:1]: 👍

File: cristin-person/src/main/java/no/unit/nva/cristin/person/create/CreateCristinPersonApiClient.java [hunk 1]
```diff
@@ -49,9 +49,11 @@ public Person createPersonInCristin(Person person, String cristinInstitutionNumb
...
     private Person createPersonFromResponse(HttpResponse<String> response) throws BadGatewayException {
         CristinPerson responseCristinPerson = getDeserializedResponse(response, CristinPerson.class);
-        Person createdPerson = responseCristinPerson.toPersonWithAuthorizedFields();
-        createdPerson.setContext(PERSON_CONTEXT);
-        return createdPerson;
+
+        return responseCristinPerson
+                   .toPersonBuilderWithAuthorizedFields()
+                   .withContext(PERSON_CONTEXT)
+                   .build();
```
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/create/CreateCristinPersonApiClient.java:56]: 👍

File: cristin-person/src/main/java/no/unit/nva/cristin/person/model/nva/Person.java
Comment (StigNorland) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/model/nva/Person.java:1]: 💯

**Review by StigNorland — APPROVED** [review]
I am biased, but I really liked the changes you made.

**Pull request merged**